simulator/tools/rofi_hal_sim.hpp

```diff
@@ -0,0 +1,138 @@
+#pragma once
+
+#include <atomic>
+#include <functional>
+#include <future>
+#include <map>
+#include <mutex>
+#include <utility>
+#include <vector>
+
+#include <gazebo/transport/transport.hh>
+
+#include "rofi_hal.hpp"
+
+#include <rofiCmd.pb.h>
+#include <rofiResp.pb.h>
+
+namespace rofi
+{
+namespace hal
+{
+namespace detail
```

Comment:
The preferred way is to use nested names, i.e., `rofi::hal::detail {` - it is shorter, more readable and require less closing brackets.